### PR TITLE
Import tests from iris.test.integration.test_grib load

### PR DESCRIPTION
**Pull request by pp-mo**
Removed from iris : https://github.com/SciTools/iris/pull/3670

### CODE REVIEW

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
+# Copyright iris-grib contributors
+#
+# This file is part of iris-grib and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Integration tests for grib2 file loading.
+
+These load various files from iris-test-data, and compare the cube with a
```
Comment (lbdreyer) [inline iris_grib/tests/integration/load_convert/test_sample_file_loads.py]:
```suggestion
These tests load various files from iris-test-data, and compare the cube with a
```

A minor change but I found this sentence a bit hard to read. I think it's not clear what you're referring to by only having "These" as the object of the sentence, especially when it is the start of a new paragraph.

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    def test_load_masked(self):
+        gribfile = tests.get_data_path(
+            ("GRIB", "missing_values", "missing_values.grib2")
+        )
+        cubes = iris.load(gribfile)
+        self.assertCML(cubes,
+                       _RESULTDIR_PREFIX + ("missing_values_grib2.cml",))
+
+    @skip_irisgrib_fails
+    def test_y_fastest(self):
```
Comment (lbdreyer) [inline iris_grib/tests/integration/load_convert/test_sample_file_loads.py]: Why do you want to keep this test? We had already removed it from the iris repo: https://github.com/SciTools/iris/pull/3632

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+# loader does not support X,Y dimensioned data (as opposed to Y,X).
+# TODO: fix this, or remove the test.
+skip_irisgrib_fails = skipIf(
+    True, "Current grib2 loader does not support this test(s)."
+)
+
+_RESULTDIR_PREFIX = ("integration", "load_convert", "sample_file_loads")
+
+
+@tests.skip_data
```
Comment (lbdreyer) [inline iris_grib/tests/integration/load_convert/test_sample_file_loads.py]: We don't typically do this for iris-grib tests as, unlike iris, we *always* download the iris-test data.
That said, maybe it's a good idea to future-proof these tests, should we decide to copy what iris does.

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py (hunk 2 above)
Comment (pp-mo) [reply iris_grib/tests/integration/load_convert/test_sample_file_loads.py]: Ok, I hadn't spotted that when I did this.

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py (hunk 3 above)
Comment (pp-mo) [reply iris_grib/tests/integration/load_convert/test_sample_file_loads.py]: I did check for usage within iris-grib and there are lots of existing uses, so I thought it best to keep it.

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py (hunk 2 above)
Comment (pp-mo) [reply iris_grib/tests/integration/load_convert/test_sample_file_loads.py]: done.

File: iris_grib/tests/integration/load_convert/test_sample_file_loads.py (hunk 3 above)
Comment (lbdreyer) [reply iris_grib/tests/integration/load_convert/test_sample_file_loads.py]: I see you are right, there are some that do use test.skip_data, but not all. We should try to be consistent with this and perhaps always include the `@test.skip_data`

**Pull request merged**